sdk/containerregistry/test-resources.json

```diff
@@ -0,0 +1,71 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "baseName": {
+            "type": "string",
+            "defaultValue": "[resourceGroup().name]",
+            "metadata": {
+                "description": "The base resource name."
+            }
+        },
+        "tenantId": {
+            "type": "string",
+            "metadata": {
+                "description": "The tenant ID to which the application and resources belong."
+            }
+        },
+        "testApplicationOid": {
+            "type": "string",
+            "metadata": {
+                "description": "The client OID to grant access to test resources."
+            }
+        },
+        "location": {
+            "type": "string",
+            "defaultValue": "[resourceGroup().location]",
+            "metadata": {
+                "description": "The location of the resource. By default, this is the same as the resource group."
+            }
+        }
+    },
+    "variables": {
+        "baseName": "[parameters('baseName')]",
+        "apiVersion": "2019-05-01",
+        "endpointValue": "[format('https://{0}.azurecr.io', parameters('baseName'))]"
+    },
+    "resources": [
+        {
+            "type": "Microsoft.ContainerRegistry/registries",
+            "apiVersion": "2019-05-01",
+            "name": "[parameters('baseName')]",
+            "location": "[parameters('location')]",
+            "properties": {        
+                "endpoint": "[variables('endpointValue')]",
+                "adminUserEnabled": true
+            },
+            "sku": {
+                "name": "Basic",
+                "tier": "Basic"
+            }
+        }
+    ],
+    "outputs": {
+        "CONTAINERREGISTRY_REGISTRY_NAME": {
+            "type": "string",
+            "value": "[variables('baseName')]"
+        },
+        "CONTAINERREGISTRY_ENDPOINT": {
+            "type": "string",
+            "value": "[variables('endpointValue')]"
+        },
+        "CONTAINERREGISTRY_USERNAME": {
+            "type": "string",
+            "value": "[listCredentials(resourceId('Microsoft.ContainerRegistry/registries', variables('baseName')), variables('apiVersion')).username]"
```

Comment:
Tip: when allocated in the same ARM template scope, you can just use the name here, which makes this much shorter and easier to read:
```json
"value": "[listCredentials(parameters('baseName'), variables('apiVersion')).username]"
```

Same below as well. See [here](https://github.com/Azure/azure-sdk-for-net/blob/04df5c5df3913015155392ef7b86f4e2133442aa/sdk/search/test-resources.json#L183) for an example.